### PR TITLE
#8068: Remove spurious docs jobs / workflows / executions

### LINKED ISSUE #8068: Reduce the number of tasks for building docs: use a single workflow
There are currently several jobs that build docs:
- Inside of `fast-dispatch-build-and-unit-tests.yaml` workflow
- Inside of fd/sd tests in scripts called `run_pre_post_commit_regressions_slow_dispatch.sh` and `run_pre_post_commit_regressions_fast_dispatch.sh`
- And the main `docs-latest-public.yaml` workflow

Centralize on a single workflow to not re-do the work multiple times.

### LINKED ISSUE #8097: Re-use built artifacts in `build-artifact.yaml` workflow in `docs-latest-public.yaml`
The `docs-latest-public.yaml` is rebuilding the libraries from scratch in the steps:
```
      - uses: ./.github/actions/install-metal-deps
        with:
          os: ubuntu-20.04
      - uses: ./.github/actions/install-metal-dev-deps
        with:
          os: ubuntu-20.04
      - name: Build tt-metal
        run: |
          export TT_METAL_HOME=$(pwd)
          make build
```

This is a duplicated effort and increases the delay before the docs are published. Re-use the existing artifacts created at an earlier step such as the one in `build-artifact.yaml`

### PR DESCRIPTION
**Pull request by dimitri-tenstorrent**
Resolve #8068 
Resolve #8097 

@TT-billteng also pushed me to get this https://github.com/tenstorrent/tt-metal/issues/8097 over the line :) 

Moved `docs-latest-public.yaml` into `all-post-commit-workflows.yaml`'

Removed the doc step from `fast-dispatch-build-and-unit-tests.yaml` and also from the secret steps hidden within `slow and fast dispatch tests (`run_pre_post_commit_regressions_fast_dispatch.sh` and `run_pre_post_commit_regressions_slow_dispatch.sh` )

### CODE REVIEW

File: .github/workflows/all-post-commit-workflows.yaml
Comment (dimitri-tenstorrent) [inline .github/workflows/all-post-commit-workflows.yaml:37]: See if you can make it depend on both sets of tests `sd-unit-tests` and `fd-unit-tests`

File: .github/workflows/all-post-commit-workflows.yaml
Comment (dimitri-tenstorrent) [inline .github/workflows/all-post-commit-workflows.yaml:9]: Are these permissions ok to have on `Post commit` workflow?

Comment (tt-rkim) [reply .github/workflows/all-post-commit-workflows.yaml:9]: I think pages is unavoidable...

We do need `id-token` unfortunately: https://github.com/actions/deploy-pages?tab=readme-ov-file#usage

File: .github/workflows/docs-latest-public.yaml
Comment (tt-rkim) [inline .github/workflows/docs-latest-public.yaml:4]: Hmm... I think we want to eventually use build artifacts for the docs here too.

I think for now this ok so we don't make this PR bigger than it needs to be. But we should probably file an issue to make sure the above is work is captured for the future

File: .github/workflows/docs-latest-public.yaml
Comment (tt-rkim) [inline .github/workflows/docs-latest-public.yaml:6]: Also why are these triggers gone? I think only the push trigger should be gone. `workflow_dispatch` is useful for those who want to build the docs on branch

**Comment by tt-rkim** [comment]
I would recommend running both post commit and the docs workflows separately and ensuring they work

File: .github/workflows/docs-latest-public.yaml
Comment (dimitri-tenstorrent) [reply .github/workflows/docs-latest-public.yaml:6]: Good point!

File: tests/scripts/run_pre_post_commit_regressions_fast_dispatch.sh
Comment (TT-billteng) [inline tests/scripts/run_pre_post_commit_regressions_fast_dispatch.sh:28]: is this script called anywhere now?

Comment (dimitri-tenstorrent) [reply tests/scripts/run_pre_post_commit_regressions_fast_dispatch.sh:28]: I am trying to call it from the docs workflow but it does not seem to work without a lot of other modifications.

Comment (TT-billteng) [reply tests/scripts/run_pre_post_commit_regressions_fast_dispatch.sh:28]: I want to still give a way for devs to build the docs locally, is this the only way now?

Comment (dimitri-tenstorrent) [reply tests/scripts/run_pre_post_commit_regressions_fast_dispatch.sh:28]: I think they can still run `./tests/scripts/run_build_docs.sh` to build docs. We do not have a tox step for docs - there are too many C++ dependencies to manage well in Tox. 

One thing I do want to do if I can is to make sure that locally and on CI the same script is executed:

i.e. locally devs call `./tests/scripts/run_build_docs.sh` and CI calls `./tests/scripts/run_build_docs.sh`

Comment (TT-billteng) [reply tests/scripts/run_pre_post_commit_regressions_fast_dispatch.sh:28]: yes good idea, otherwise it may be broken if we don't use it in CI

**Comment by dimitri-tenstorrent** [comment]
Build Docs - https://github.com/tenstorrent/tt-metal/actions/runs/8941441478
All Commits - https://github.com/tenstorrent/tt-metal/actions/runs/8941437613

File: .github/workflows/all-post-commit-workflows.yaml
Comment (TT-billteng) [inline .github/workflows/all-post-commit-workflows.yaml:37]: do you need to wait for these steps to finish? Given that you still need to compile right now on GH runners, the post-commit time is now quite a bit longer (1h 24min in your run)

**Review by TT-billteng — CHANGES_REQUESTED** [review]
let's not submit this until the post-commit time is back to normal; starting docs step after tests increases post-commit time by quite a bit

File: .github/workflows/all-post-commit-workflows.yaml
Comment (dimitri-tenstorrent) [reply .github/workflows/all-post-commit-workflows.yaml:37]: I can revert not to block it on the successful tests, but then we would be publishing the docs even if the tests are failing. I'll make the change to not gate on the tests.

**Comment by dimitri-tenstorrent** [comment]
> let's not submit this until the post-commit time is back to normal; starting docs step after tests increases post-commit time by quite a bit

Many many thanks for the review!

I have updated to use the build runner for docs and removed the gate in all_commit tests on the unit-tests.
Here are the successful runs of all-commit and build-docs workflows:
- https://github.com/tenstorrent/tt-metal/actions/runs/8950827437
- https://github.com/tenstorrent/tt-metal/actions/runs/8950828950

You can see that the build of docs take 6 mins.

File: .github/workflows/all-post-commit-workflows.yaml
Comment (TT-billteng) [reply .github/workflows/all-post-commit-workflows.yaml]: think we can keep this check after it uses the build artifact

**Review by TT-billteng — APPROVED** [review]
Looks so much cleaner, thanks !

**Comment by dimitri-tenstorrent** [comment]
I am going to test the standalone workflow after the merge since it is a new one and it does not yet appear in the list of available workflows.

**Review by TT-billteng — APPROVED** [review]

**Pull request merged**
